applications/GeoMechanicsApplication/custom_constitutive/interface_coulomb_with_tension_cut_off.cpp

```diff
@@ -0,0 +1,219 @@
+        << "value of " << rVariable.Name() << " for property " << rMaterialProperties.Id()
+        << " is out of range: " << rMaterialProperties[rVariable] << " is not in [0.0, "
+        << (MaxValue ? std::to_string(*MaxValue) + "]" : "->") << std::endl;
+}
+
+} // namespace
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer InterfaceCoulombWithTensionCutOff::Clone() const
+{
+    auto p_result                      = std::make_shared<InterfaceCoulombWithTensionCutOff>(*this);
+    p_result->mTractionVector          = mTractionVector;
+    p_result->mTractionVectorFinalized = mTractionVectorFinalized;
+    p_result->mRelativeDisplacementVectorFinalized = mRelativeDisplacementVectorFinalized;
+    p_result->mCoulombWithTensionCutOffImpl        = mCoulombWithTensionCutOffImpl;
+    return p_result;
+}
+
+Vector& InterfaceCoulombWithTensionCutOff::GetValue(const Variable<Vector>& rVariable, Vector& rValue)
+{
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+        rValue = mTractionVector;
+    } else {
+        rValue = ConstitutiveLaw::GetValue(rVariable, rValue);
+    }
+    return rValue;
+}
+
+void InterfaceCoulombWithTensionCutOff::SetValue(const Variable<Vector>& rVariable,
+                                                 const Vector&           rValue,
+                                                 const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+        mTractionVector = rValue;
+    } else {
+        KRATOS_ERROR << "Can't set value of " << rVariable.Name() << ": unsupported variable\n";
+    }
+}
+
+SizeType InterfaceCoulombWithTensionCutOff::WorkingSpaceDimension() { return N_DIM_2D; }
+
+int InterfaceCoulombWithTensionCutOff::Check(const Properties&   rMaterialProperties,
+                                             const GeometryType& rElementGeometry,
+                                             const ProcessInfo&  rCurrentProcessInfo) const
+{
+    const auto result = ConstitutiveLaw::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+
+    CheckProperty(rMaterialProperties, GEO_COHESION);
+    CheckProperty(rMaterialProperties, GEO_FRICTION_ANGLE);
+    CheckProperty(rMaterialProperties, GEO_DILATANCY_ANGLE, rMaterialProperties[GEO_FRICTION_ANGLE]);
+    CheckProperty(rMaterialProperties, GEO_TENSILE_STRENGTH,
+                  rMaterialProperties[GEO_COHESION] /
+                      std::tan(MathUtils<>::DegreesToRadians(rMaterialProperties[GEO_FRICTION_ANGLE])));
+    CheckProperty(rMaterialProperties, INTERFACE_NORMAL_STIFFNESS);
+    CheckProperty(rMaterialProperties, INTERFACE_SHEAR_STIFFNESS);
+    return result;
+}
+
+ConstitutiveLaw::StressMeasure InterfaceCoulombWithTensionCutOff::GetStressMeasure()
+{
+    return ConstitutiveLaw::StressMeasure_Cauchy;
+}
+
+SizeType InterfaceCoulombWithTensionCutOff::GetStrainSize() const
+{
+    return VOIGT_SIZE_2D_INTERFACE;
```

Comment:
Let's add a comment here, too:
```suggestion
    // Note that this implementation assumes line interface elements. It needs to be modified when planar interface elements become available.
    return VOIGT_SIZE_2D_INTERFACE;
```